include/deal.II/dofs/dof_handler.h

```diff
@@ -713,13 +713,15 @@ class DoFHandler : public Subscriptor
    * function set_fe().
    */
   void
-  distribute_dofs(const FiniteElement<dim, spacedim> &fe);
+  distribute_dofs(const FiniteElement<dim, spacedim>    &fe,
+                  const dealii::types::global_dof_index &virtual_dofs = 0);
```

Comment:
Do I understand correctly that these should be interpreted as the *total* number of virtual dofs? In your example, you are making all of them locally owned by all processors. This does not sound right to me, i.e., in the example above, if you run on *n* processors you should end up with a matrix that has `n` on the diagonal on the virtual dofs block.
Maybe we should add a constructor that actually takes an `IndexSet` as virtual dofs.

Reply:
@luca-heltai Yes right now I only explored what this parameter should mean for serial and parallel::distributed (and not for the others). We'll have to discuss how the interface should look like anyway.

I really don't like this additional argument for `distribute_dofs` and I think it should either be a part of the finite element or a different method/interface.